html/canvas/color.window.js

```diff
@@ -0,0 +1,64 @@
+const canvas = document.body.appendChild(document.createElement("canvas"));
+const context = canvas.getContext("2d");
+
+[
+  {
+    "input": "rgb(255, 0, 255)",
+    "output": "#ff00ff"
+  },
+  {
+    "input": "rgba(255, 0, 255, 1)",
+    "output": "#ff00ff"
+  },
+  {
+    "input": "#ff00ffff",
+    "output": "#ff00ff"
+  },
+  {
+    "input": "#ff00ff00",
+    "output": "rgba(255, 0, 255, 0)"
+  },
+  {
+    "input": "rgba(255, 0, 255,    0)",
+    "output": "rgba(255, 0, 255, 0)"
+  },
+  {
+    "input": "color(srgb 1 0 1)",
+    "output": "#ff00ff"
```

Comment:
My preference would be that this serialize as `color(srgb 1 0 1)`. That is, any of the newer color types (color(), lab(), ...), serialize as they would in CSS for a computed value serialization. I believe this is what @svgeesus was advocating in https://github.com/w3c/csswg-drafts/issues/10550 as well.

The reason for this preference is that it sets the precedent that we are preserving the existing behavior for existing syntax, but allowing consistency with CSS for any newer syntaxes going forward.

I think the test could also use test cases for a couple other oddballs:

  space separated rgb/rgba
     (I assume these would be serialized as hex / rgba depending on alpha)
  rgb/rgba with values less than 0 and greater than 255 
     (I assume these would clamped to 0-255 at parse time and serialized as hex / rgba depending on alpha.
  comma separated hsl/hsla
     (I assume these would be converted to rgb and serialized as hex / rgba depending on alpha)
  space separated hsl/hsla
     (I assume these would be converted to rgb and serialized as hex / rgba depending on alpha)
  space separated rgb/rgba/hsl/hsla with a parameter set to none
     (I assume these would have the none converted to 0 at serialization time and serialized as hex / rgba depending on alpha).

Reply:
Yes, in Servo `color(srgb 1 0 1)` doesn't have the legacy srgb flag, so in CSS it serializes as is, not with `rgb()` or `rgba()`. Then in canvas we serialize as is too, matching CSS. Seems similar for Blink and WebKit.

Reply:
> Presumably `transparent` becoming `rgba()` is still okay?

Yes, [CSS Color 4 says](https://drafts.csswg.org/css-color-4/#serializing-sRGB-values)

> Thus, the serialized declared value of `transparent` is the string "transparent", while the serialized computed value of transparent is the string "rgba(0, 0, 0, 0)".